routers/web/repo/view.go

```diff
@@ -98,28 +92,22 @@ func findReadmeFileInEntries(ctx *context.Context, entries []*git.TreeEntry) (*n
 		}
 		if i, ok := util.IsReadmeFileExtension(entry.Name(), exts...); ok {
 			log.Debug("Potential readme file: %s", entry.Name())
-			if readmeFiles[i] == nil || base.NaturalSortLess(readmeFiles[i].name, entry.Blob().Name()) {
-				name := entry.Name()
-				isSymlink := entry.IsLink()
+			if readmeFiles[i] == nil || base.NaturalSortLess(readmeFiles[i].Name(), entry.Blob().Name()) {
 				target := entry
-				if isSymlink {
+				if entry.IsLink() {
 					var err error
 					target, err = entry.FollowLinks()
 					if err != nil && !git.IsErrBadLink(err) {
-						return nil, err
+						return "", nil, err
 					}
 				}
 				if target != nil && (target.IsExecutable() || target.IsRegular()) {
-					readmeFiles[i] = &namedBlob{
-						name,
-						isSymlink,
-						target.Blob(),
-					}
+					readmeFiles[i] = target
```

Comment:
The logic here has been changed. The previous name comes from the original file, not the target file.

Reply:
Oh I see. That's subtle. With a symlink in the repo like this:

```
sdffsf$ tree .github/
.github/
├── a
│   └── b
│       └── c
│           └── _hocs.md
└── README.md -> a/b/c/_hocs.md

3 directories, 2 files
```

6976d703d5776e556b8644082468aff2a51f484a renders with the nonexistent path ".github/_hocs.md"

![Screenshot 2023-02-13 at 22-38-33 sdffsf](https://user-images.githubusercontent.com/987487/218633179-7bd92bf3-0441-4e56-b955-0e07d62d3cbd.png)

I've moved (d9393ee3bb1ae5f8fbc9225dc71e7479195a6ae4) the symlink checking logic into renderReadmeFile() and now it renders like it used to:

![Screenshot 2023-02-13 at 22-37-32 sdffsf](https://user-images.githubusercontent.com/987487/218633174-11097667-2617-4bab-bf1e-51bad6af349e.png)